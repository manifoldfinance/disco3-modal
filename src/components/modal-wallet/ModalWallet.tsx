import React from 'react';
import styled from 'styled-components';
import { useWeb3React, UnsupportedChainIdError } from '@web3-react/core';

import { Wallet } from '../wallet';
import { Modal } from './Modal';
import Close from './x.svg';

interface IModalWalletProps {
  isOpen: boolean;
  onClose: () => void;
}

const Content = styled.div`
  display: flex;
  flex-flow: column nowrap;
  margin: 0;
  padding: 0;
  width: 100%;
`;

const Header = styled.div`
  position: relative;

  h5 {
    margin: 0;
    margin-bottom: 0.5rem;
    font-size: 1rem;
    font-weight: 400;
  }

  h5:last-child {
    margin-bottom: 0px;
  }

  h4 {
    margin-top: 0;
    font-weight: 500;
  }
`;

const HeaderRow = styled.div`
  display: flex;
  flex-flow: row nowrap;
  padding: 1rem 1rem;
  font-weight: 500;
  padding: 1rem;
`;

const CloseIcon = styled.div`
  position: absolute;
  right: 1rem;
  top: 14px;
  &:hover {
    cursor: pointer;
    opacity: 0.6;
  }
`;

const CloseColor = styled.img`
  path {
    stroke: ${({ theme }) => theme.text4};
  }
`;

const Title = styled.div``;

export const ModalWallet: React.FC<IModalWalletProps> = ({ isOpen, onClose }) => {
  const { error } = useWeb3React();

  return (
    <Modal isOpen={isOpen} onDismiss={onClose} minHeight={false} maxHeight={90}>
      <Content>
        <Header>
          <HeaderRow>
            <CloseIcon onClick={onClose}>
              <CloseColor src={Close} alt="close icon" />
            </CloseIcon>
            {!error && <Title>Connect to a wallet</Title>}
            {error && (
              <Title>
                {error instanceof UnsupportedChainIdError ? 'Wrong Network' : 'Error connecting'}
              </Title>
            )}
          </HeaderRow>
        </Header>
        <Wallet />
      </Content>
    </Modal>
  );
};
