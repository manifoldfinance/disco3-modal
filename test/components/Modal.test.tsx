import { Modal } from '../../src/components/modal-wallet/Modal';

describe('Modal tests', () => {
  test('it works', () => {
    expect(Modal).toMatchSnapshot();
  });
});
