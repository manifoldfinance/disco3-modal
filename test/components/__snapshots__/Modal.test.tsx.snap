// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests it works 1`] = `[Function]`;
